### ISSUE TITLE
Possibility to check for HTTPs with self-signed certificates

### ISSUE THREAD

**Issue by xorguy** [louislam/uptime-kuma#15]
HI.

It will be handy to be able to override and check for servers (HTTPS) with self signed certificates.

Thanks.

**Comment by fire1ce** [comment]
This should be top priority for the self hosted community

**Comment by Expro** [comment]
Definitely. Ability to add custom CA and certificates would be very valuable. Or, as workaround, ability to ignore certificate errors,

**Comment by joe-eklund** [comment]
> Definitely. Ability to add custom CA and certificates would be very valuable. Or, as workaround, ability to ignore certificate errors,



Fwiw, you can inject your own CA or ignore all SSL cert issues by setting some environment variables. This can be accomplished by doing the following (note this is a docker stack file for docker swarm, but docker-compose is close):

```
version: "3.8"
services:
  uptime_kuma:
    image: louislam/uptime-kuma
    volumes:
       - /uptime_kuma/data:/app/data
       - /uptime_kuma/certs:/certs
    deploy:
      restart_policy:
        condition: any
    ports:
      - 3001:3001
    environment:
      - NODE_EXTRA_CA_CERTS=/certs/ca-bundle.crt # inject your CA.
      #-NODE_TLS_REJECT_UNAUTHORIZED=0 # ignore ssl issues, potentially dangerous but useful for testing :)
```

@louislam maybe you could add some documentation in the readme or wiki on how to do this since it seems to be a common question?

**Comment by louislam** [comment]
Implemented in 1.0.8

 https://github.com/louislam/uptime-kuma/releases/tag/1.0.8
